### ISSUE TITLE
Better documentation

### ISSUE THREAD

**Issue by dwrolvink** [hikerpig/note-graph#24]
Hi, an end-user of mine asked me if I could add this package as a dependency because he likes the way it looks.

It looks like a nice fancy graph system, but I when I try to do things I have to dive into the code to find out how I can do things. And I am not a typescript person so this is a pretty painful process for me.

For example, to find out how to have the graph not bork itself into a blur if it is not 100% of screen width and height, I had to dive into the code to find out that I can just do
``` js
        graphView = new NOTE_GRAPH.NoteGraphView({
            container: args.graph_container,
            graphModel,
            width: <num px>,
            height: <num px>
        })
```

And to select a node, again, look into the code. For reference, this works as I've found out through trial and error, still not sure if this is the correct way to do it
``` js
        graphView.actions.selectNode(graphView.model, parseInt(args.current_node_graph_id), false)
        graphView.updateViewModeInteractiveState()
```
etc.

Now I have the issue that the nodes are wayy too close together to be useable in a big graph. I'm sure you have something nice for this, but it's not clear from the get-go.

tl;dr: nice project, good luck with building it out further, but do not forget the documentation aspect, it is very useful for new people like me to quickly pick it up and run with it.

Cheers.

**Comment by hikerpig** [comment]
@dwrolvink Thank you for this valuable advice.

Basicly note-graph is a glue layer between the underlying [force-graph](https://github.com/vasturiano/force-graph) lib and my extra abstractions for [gatsby-project-kb](https://gatsby-project-kb.vercel.app/features/graph-view). 

I didn't pay attention to the docs then. Storybook is capable of displaying the demo and the exact story source code of it .  But as you say, the documentation is indeed pretty poor. I will try to add more detail instructions about the options, and add more  usage examples.

By 'wavy too close' , do you mean something like this ? https://note-graph-hikerpig.vercel.app/?path=/story/other-datasets--foam

This is because by default note-graph adjusts the scaling so that all nodes are in the visible area, which in a graph with many nodes can lead to the overly dense situation you describe.

<img width="1003" alt="image" src="https://user-images.githubusercontent.com/2259688/164373481-a67d303a-e9bf-40c4-aca6-788e484d5711.png">

**Comment by dwrolvink** [comment]
Ah that makes sense, he got it from the gatsby project actually.

Yes, that would be the problem then. Do you have any insight in how the zooming works (in this system or in force-graph?) the ideal would be to zoom in far enough to see a selected node and all its direct links. (If you don't that's fine too, it seems to be a bit more advanced of a usecase, but maybe you have figured it out already).

I was thinking yesterday, maybe having this project be used by many other people isn't really in your scope, in that case this issue is a bit of a mute point. But anyways, some feedback nonetheless :smile: 

You can close this issue if you want.

**Comment by hikerpig** [comment]
@dwrolvink Haha of course I want more people to use my lib, but I guess this notes plus force-graph scenario is not so common. And developers may want to develop their own abstractions instead.

Back to the zoom problem you met.
- When calling `graphView.setSelectedNodes(nodeIds, { shouldZoomToFit: true })`, note-graph will call `this.forceGraph.zoomToFit` with some filters and try to zoom to a proper scale where all selected nodes are visible.
- Check `zoomToFit` function docs in [force-graph project README](https://github.com/vasturiano/force-graph)


Example of this `setSelectedNodes` can be found in [this story](https://note-graph-hikerpig.vercel.app/?path=/story/example--set-selected-nodes-and-zoom)
You can click the 'Docs' panel and 'Show Code' button of the example to see example code.
![image](https://user-images.githubusercontent.com/2259688/164457120-001fc225-a82a-44cf-b98a-a2f5b9d9b363.png)

**Comment by dwrolvink** [comment]
Nice! I had missed the Show Code part. For now we've opted to limit ourselves to force-graph by itself, but this will be useful if we get another feature request for note-graph.

I'll close this now as for us there is no longer an open feature request. Cheers!
